.github/workflows/self-scheduled.yml

```diff
@@ -297,6 +297,56 @@ jobs:
           name: ${{ matrix.machine_type }}_run_tests_torch_cuda_extensions_gpu_test_reports
           path: /workspace/transformers/reports/${{ matrix.machine_type }}_tests_torch_cuda_extensions_gpu
 
+  run_tests_quantization_torch_gpu:
+    name: Quantization tests
+    strategy:
+      fail-fast: false
+      matrix:
+        machine_type: [single-gpu, multi-gpu]
+    runs-on: ['${{ matrix.machine_type }}', nvidia-gpu, t4, daily-ci]
+    container:
+      image: huggingface/transformers-quantization-latest-gpu
+      options: --gpus all --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
+    needs: setup
+    steps:
+      - name: Update clone
+        working-directory: /transformers
+        run: git fetch && git checkout ${{ github.sha }}
+
+      - name: Reinstall transformers in edit mode (remove the one installed during docker image build)
+        working-directory: /transformers
+        run: python3 -m pip uninstall -y transformers && python3 -m pip install -e .
```

Comment:
@ydshieh I do not understand why we don't:
1. Install from main on the docker build with `pip install git+https://github.com/hugginface/transformers@main` (and get all the necessary dependencies)
2. `pip uninstall -y transformers` in the docker 
3. `actions/checkout@v3`
4. `pip install -e  .`
(As discussed offline with @younesbelkada this should come in a separate PR)

Reply:
There is no deep reason for what we are doing currently: we just install `transformers` during docker image build and reuse it.

What you suggest (by installing `main` during docker image build) has a consequence that `we can't do experiment easily in a branch when we change `setup.py`).

Also by using `actions/checkout@v3`, the path to `transformers` will be different from the current approach, and therefore we have to change all the working directory path and artifact path, otherwise the report will be empty.